R/slide.R

```diff
@@ -459,25 +459,25 @@ epi_slide_opt <- function(x, col_names, f, ..., before, after, ref_time_values,
   # `data.table` and `slider` (or a function that has the exact same
   # definition, e.g. if the function has been reexported or defined
   # locally).
-  if (any(sapply(
-    c(frollmean, frollsum, frollapply),
+  if (any(map_lgl(
+    list(frollmean, frollsum, frollapply),
     function(roll_fn) {
-      isTRUE(identical(f, roll_fn))
+      identical(f, roll_fn)
     }
   ))) {
     f_from_package <- "data.table"
-  } else if (any(sapply(
-    c(slide_sum, slide_prod, slide_mean, slide_min, slide_max, slide_all, slide_any),
+  } else if (any(map_lgl(
+    list(slide_sum, slide_prod, slide_mean, slide_min, slide_max, slide_all, slide_any),
     function(roll_fn) {
-      isTRUE(identical(f, roll_fn))
+      identical(f, roll_fn)
     }
   ))) {
     f_from_package <- "slider"
   } else {
     # `f` is from somewhere else and not supported
     cli_abort(
       c(
-        "slide function `f` is not supported",
+        "problem with {rlang::expr_label(rlang::enexpr(f))}",
```

Comment:
?!?! check this out:

```r
jhu_csse_daily_subset %>%
  group_by(geo_value) %>%
  epi_slide_opt(
    cases,
    f = function(...) {"pretend";"this";"is";"real";"function";data.table::frollsum(...)}, new_col_name = "cases_7dav", names_sep = NULL, before = 3, after = 3
  ) %>%
  ungroup()
#> Error in `epi_slide_opt()` at dplyr/R/group-by.R:153:3:
#> ! problem with `function (...) { "pretend" "this" "is" "real" "function"
#>   data.table::frollsum(...) }`
#> ℹ `f` must be one of `data.table`'s rolling functions (`frollmean`, `frollsum`, `frollapply`. See
#>   `?data.table::roll`) or one of `slider`'s specialized sliding functions (`slide_mean`, `slide_sum`,
#>   etc. See `?slider::`summary-slide`` for more options).
#> Run `rlang::last_trace()` to see where the error occurred.

complain <- function(f) {
  cli_abort(
    c(
      "problem with {rlang::expr_label(rlang::enexpr(f))}",
      "i" = "`f` must be one of `data.table`'s rolling functions (`frollmean`,
            `frollsum`, `frollapply`. See `?data.table::roll`) or one of
            `slider`'s specialized sliding functions (`slide_mean`, `slide_sum`,
            etc. See `?slider::\`summary-slide\`` for more options)."
    ),
    class = "epiprocess__epi_slide_opt__unsupported_slide_function",
    epiprocess__f = f
  )
}
complain(function(...) {"pretend";"this";"is";"real";"function";data.table::frollsum(...)})
#> Error in `complain()`:
#> ! problem with `function(...) ...`
#> ℹ `f` must be one of `data.table`'s rolling functions (`frollmean`, `frollsum`, `frollapply`. See
#>   `?data.table::roll`) or one of `slider`'s specialized sliding functions (`slide_mean`, `slide_sum`,
#>   etc. See `?slider::`summary-slide`` for more options).
#> Run `rlang::last_trace()` to see where the error occurred.
```
I was hoping we'd get the latter.  But I guess former is okay.

Reply:
Oh, maybe `caller_arg(f)` can be used to make this less messy.

Reply:
Changed in https://github.com/cmu-delphi/epiprocess/pull/436/commits/fc2590d7401d98598ec58aba3609f64d0ac88f00. 

First example now returns
```
Error in `epi_slide_opt()`:
! problem with "function(...) ..."
ℹ `f` must be one of `data.table`'s rolling functions (`frollmean`, `frollsum`, `frollapply`. See `?data.table::roll`) or one of `slider`'s
  specialized sliding functions (`slide_mean`, `slide_sum`, etc. See `?slider::`summary-slide`` for more options).
Run `rlang::last_trace()` to see where the error occurred.
```